connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -1001,6 +1050,16 @@ object SparkSession extends Logging {
     }
 
     private def applyOptions(session: SparkSession): Unit = {
+      // Only attempts to set Spark SQL configurations.
+      // If the configurations are static, it might throw an exception so
+      // simply ignore it for now.
+      initOptions
```

Comment:
I don't think this is needed. You are setting the initOptions when starting the driver. Those will be automatically propagated to any connect session.

Reply:
It actually could matter if users do, for example,

```bash
spark-submit ... --remote sc://localhost --conf=...
```

and the server is already running somewhere.